src/formatDate.js

```diff
@@ -50,7 +50,61 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const formated = [];
+  const splitDate = date.split(fromFormat[3]);
+  let year;
+  let month;
+  let day;
+
+  for (let i = 0; i < fromFormat.length; i++) {
+    if (fromFormat[i] === 'YYYY' || fromFormat[i] === 'YY') {
+      year = splitDate[i];
+    }
+
+    if (fromFormat[i] === 'MM') {
+      month = splitDate[i];
+    }
+
+    if (fromFormat[i] === 'DD') {
+      day = splitDate[i];
+    }
+  }
+
+  for (const toForm of toFormat) {
+    switch (toForm) {
+      case ('DD'):
+        formated.push(day);
+
+        break;
+
+      case ('MM'):
+        formated.push(month);
+
+        break;
+
+      case ('YYYY'):
+        if (year.length === 4) {
+          formated.push(year);
+        } else if (year < 30) {
+          formated.push(20 + year);
+        } else {
+          formated.push(19 + year);
```

Comment:
4, 19, 20 and 30 are the magic numbers, so make sense to move the to the constants